src/pyscaffoldext/dsproject/templates/Dockerfile.template

```diff
@@ -0,0 +1,55 @@
+# Dockerfile
+# * builder steps:
+#  - create environment and build package
+#  - save environment for runner
+# * runner steps:
+#  - recreate the same environment and install built package
+#  - optionally execute provided console_script in ENTRYPOINT
+#
+# Alternatively, remove builder steps, take `environment.docker.yml` from your repo
+# and `pip install ${name}` using an artifact store. Faster and more robust.
+
+FROM condaforge/miniforge3 AS builder
+WORKDIR /tmp
+
+COPY . /tmp
+
+RUN conda install mamba -n base
+RUN mamba env create -f environment.yml
+# RUN pip install -e . # not needed since it's in environment.yml
+SHELL ["mamba", "run", "-n", "${name}", "/bin/bash", "-c"]
+
+# build package
+RUN tox -e build
+RUN mamba env export -n ${name} -f environment.docker.yml
+
+FROM condaforge/miniforge3 AS runner
+WORKDIR /app
+
+# Copy caches, environment.docker.yml and built package
+COPY --from=builder /root/.cache/pip /root/.cache/pip
```

Comment:
If we switch to the user `app`, this cache (`root/.cache/pip`) will lose it effectiveness, right?